Sources/SPMTestSupport/MockWorkspace.swift

```diff
@@ -510,13 +510,16 @@ public final class MockWorkspace {
         public let diagnostics: [Basics.Diagnostic]
     }
 
-    public func checkPrecomputeResolution(_ check: (ResolutionPrecomputationResult) -> Void) throws {
```

Comment:
This method should be full async. It makes more sense to have it just return the `ResolutionPrecomputationResult` result and let the user do the checking inline